### PR TITLE
d_a_obj_xfuta match

### PR DESCRIPTION
**Pull request by poketannenbaum**

### CODE REVIEW

File: include/d/actor/d_a_obj_xfuta.h [hunk 1]
```diff
@@ -1,25 +1,47 @@
...
-        BOOL _execute();
-        BOOL _draw();
-    
-    public:
-        /* 0x290 */ void* vtbl;
-        /* 0x294 */ request_of_phase_process_class mPhs;
-        /* 0x29C */ J3DModel* mpModel;
-    };
-};
+static int tev_mode[] = {TEV_TYPE_ACTOR,
```
Comment (magcius) [inline include/d/actor/d_a_obj_xfuta.h:9]: This should go inside the .cpp file.

File: include/d/actor/d_a_obj_xfuta.h [hunk 2]
```diff
@@ -1,25 +1,47 @@
...
+                         TEV_TYPE_BG0,
+                         TEV_TYPE_BG1,
+                         TEV_TYPE_BG2,
+                         TEV_TYPE_BG3,
+                         TEV_TYPE_BG0_FULL,
+                         TEV_TYPE_BG1_FULL,
+                         TEV_TYPE_BG2_FULL,
+                         TEV_TYPE_BG3_FULL,
+                         TEV_TYPE_PLAYER,
+                         0x5B,
```
Comment (magcius) [inline include/d/actor/d_a_obj_xfuta.h:19]: I'll fix these enums after the fact; for now this is fine.

File: src/d/actor/d_a_obj_xfuta.cpp [hunk 3]
```diff
@@ -4,79 +4,119 @@
...
+    cXyz pos;
+    dDemo_manager_c* demo = dComIfGp_demo_get();
+    if (demo->getMode() != 1) {
+        pos.x = 0.0;
+        pos.y = 0.0;
+        pos.z = 0.0;
+        g_env_light.settingTevStruct(tev_mode[1], &pos, &tevStr);
+        dComIfGd_setListBG();
+        g_env_light.setLightTevColorType(mpModel, &tevStr);
+        J3DModelData* mdl_data = mpModel->getModelData();
```
Comment (magcius) [inline src/d/actor/d_a_obj_xfuta.cpp:84]: Is this getModelData necessary?

File: src/d/actor/d_a_obj_xfuta.cpp [hunk 4]
```diff
@@ -4,79 +4,119 @@
...
+    return TRUE;
 }
 
 /* 00000338-0000040C       .text _draw__Q210daObjXfuta5Act_cFv */
-BOOL daObjXfuta::Act_c::_draw() {
-    /* Nonmatching */
+bool daObjXfuta::Act_c::_draw() {
+    cXyz pos;
+    dDemo_manager_c* demo = dComIfGp_demo_get();
+    if (demo->getMode() != 1) {
```
Comment (magcius) [inline src/d/actor/d_a_obj_xfuta.cpp:77]:
```suggestion
    if (dComIfGp_demo_mode() == 1)
        return;

    cXyz pos(0.0f, 0.0f, 0.0f);
```

**Comment by LagoLunatic** [comment]
You can update xfuta in configure.py to be "Matching" instead of "NonMatching". Then the system will automatically check it to be sure it's correct.

**Comment by magcius** [comment]
You still need to move tev_mode out of the header file.

**Pull request merged**